File: Python/Completed Problems/Problem 2/Problem 2.py
Fibonacci = [1,1,2]

sum = 0
 
n=2
while Fibonacci[n] <= 4000000:
    Fibonacci.append(Fibonacci[n] + Fibonacci[n-1])
    n+=1

for item in Fibonacci:
    if item <= 4000000 and item%2 == 0:
        sum+=item

print(Fibonacci)
print(sum)

File: Haskell/In Progress/Problem 2/Problem2.hs
fibonacci :: (Integral a) => a -> a
fibonacci n
    | n <= 1    = 1
    | otherwise = (fibonacci (n - 1)) + (fibonacci (n - 2))

let fibonacciList = [fibonacci x | x <- [0..]]

main = putStrLn (sum [fibonacci x | x <- [0..], (fibonacci x) <= 4000000, ((fibonacci x) `mod` 2 == 0)])
